src/gen_smtp_server_session.erl

```diff
@@ -874,6 +923,32 @@ check_bare_crlf(Binary, _Prev, Op, Offset) ->
 			end
 	end.
 
+ranch_handshake(gte16, Ref, _Sock) ->
+	ranch:handshake(Ref);
+ranch_handshake(lt16, Ref, Sock) ->
+	ok = ranch:accept_ack(Ref),
+	{ok, Sock}.
+
+send(#state{transport = Transport, socket = Sock}, Data) ->
+    ok = Transport:send(Sock, Data).
```

Comment:
I'm often getting `@gen_smtp_server_session:send:933 gen_server <0.9016.2> terminated with reason: no match of right hand value {error,closed} in gen_smtp_server_session:send/2 line 933` in production. I believe it's because SMTP server clients are closing the socket without waiting for response from us. It's kind of ok, but maybe we should handle such cases explicitly? Because now it spams the logs.

Before this PR we were just ignoring such errors by not matching on smtp_socket:send return values

Reply:
If we received all info and did report the success to the remote client, then it is ok that remote drops the connection without properly saying goodbye.  So our behavior here really depends on the state we are in, if we got success with receiving the message then we should ignore the send-error, if we are in the middle of the send hand shake then we could log something and also give up. In the end it is up to the remote to try again.

Reply:
@mworrell another option is to define some `terminate` optional behavior callback

Reply:
Good idea, then we can log in the `terminate` if we are terminating in the middle of some transaction or just be silent if all was received and well.

The send code could just stop the `gen_server` if an error occurs.

Reply:
I tried to fix this, but it seems to require some major rewrite (eg, get rid of separate process to receive message body). I think I will revert this `ok = ` pattern for now (so it will behave the same as before my PR) and will make a separate PR to clean this up later.
What do you think?

Reply:
So yep, I disabled send result checks for now and will add it in another PR